packages/web/src/components/track/GiantTrackTile.tsx

```diff
@@ -538,7 +551,17 @@ export const GiantTrackTile = ({
                   badgeSize={18}
                   popover
                 />
-              </div>
+                {isEditAlbumsEnabled &&
+                album?.playlistName &&
+                album.permalink ? (
+                  <>
+                    <span>from</span>
+                    <TextLink variant='visible' textVariant='display' asChild>
+                      <Link to={album.permalink}>{album.playlistName}</Link>
```

Comment:
So cool! Album or "collection"?

Reply:
Would we just need to check for album existence rather than name + permalink?

Reply:
Album because we pass `collectionType: 'album'` into the query above

I'll fix the check. good call.